core/templates/dev/head/pages/exploration_player/RefresherExplorationConfirmationModalService.js

```diff
@@ -35,7 +36,8 @@ oppia.factory('RefresherExplorationConfirmationModalService', [
             function($scope, $uibModalInstance, $window) {
               $scope.confirmRedirect = function() {
                 redirectConfirmationCallback();
-                var url = '/explore/' + refresherExplorationId;
+                var url = '/explore/' + refresherExplorationId +
+                  currentQueryString;
```

Comment:
Nope -- please extract the collection_id field explicitly if it exists, and add it manually.

Also could you use UrlInterpolationService to construct the initial URL?

Reply:
I did that initially, but this was required for multiple chain redirections.

Reply:
Yep, you should extract the redirect parameter explicitly as well.

Reply:
Oh, ok, so I'll extract parent and collection_id fields manually and add it back.

Reply:
Also, refresher explorations won't be part of collections right?

Reply:
Nope, I don't think so!

Reply:
Done